engine/src/test/java/org/camunda/bpm/engine/test/api/identity/IdentityServiceTest.java

```diff
@@ -333,20 +334,31 @@ public void findGroupByUnexistingId() {
   public void testCreateMembershipUnexistingGroup() {
     User johndoe = identityService.newUser("johndoe");
     identityService.saveUser(johndoe);
-
-    thrown.expect(ProcessEngineException.class);
-
-    identityService.createMembership(johndoe.getId(), "unexistinggroup");
+    try {
+      identityService.createMembership(johndoe.getId(), "unexistinggroup");
+      fail("NullValueException is expected");
+    } catch(Exception ex) {
+      if(!(ex instanceof NullValueException)) {
+        fail("NullValueException is expected  but other exception was received: " + ex);
+      }
+      assertEquals("No group found with id 'unexistinggroup'.: group is null", ex.getMessage());
+    }
```

Comment:
Instead of the complex try, catch and instanceof check, you could simply expect an exception of type `NullValueException` with the `ExpectedExceptionRule`. You can also use this rule to expect a certain exception message.

Reply:
Example: https://github.com/camunda/camunda-bpm-platform/blob/master/engine/src/test/java/org/camunda/bpm/engine/test/api/identity/IdentityServiceTest.java#L367-L373

Reply:
@mboskamp thanks for your inputs. will do the changes.

Reply:
@mboskamp done!